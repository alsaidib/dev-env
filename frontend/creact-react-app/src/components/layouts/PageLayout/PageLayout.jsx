import React from 'react';
import PropTypes from 'prop-types';

import { layout } from './style.module.scss';

const PageLayout = ({ children, style }) => (
  <div className={layout} style={style}>
    {children}
  </div>
);
PageLayout.propTypes = {
  children: PropTypes.any,
  style: PropTypes.object,
};

export default React.memo(PageLayout);
