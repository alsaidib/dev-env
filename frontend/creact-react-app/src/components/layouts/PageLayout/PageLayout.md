## Default

```jsx { "props": { "style": { "width": 50, "height": 50 } } }
<PageLayout style={{ width: 150, height: 50 }}>Content</PageLayout>
```
