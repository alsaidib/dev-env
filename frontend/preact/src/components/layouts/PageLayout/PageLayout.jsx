import { layout } from './style.module.css';

const Layout = ({ children }) => <div className={layout}>{children}</div>;

export default Layout;
