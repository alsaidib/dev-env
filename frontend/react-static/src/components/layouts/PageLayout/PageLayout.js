import React from 'react';
import PropTypes from 'prop-types';

import { layout } from './style.module.scss';

const Layout = ({ children }) => <div className={layout}>{children}</div>;
Layout.propTypes = {
  children: PropTypes.any,
};

export default React.memo(Layout);
