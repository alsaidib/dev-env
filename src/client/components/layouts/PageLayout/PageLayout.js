import React from 'react';
import PropTypes from 'prop-types';

import styles from './style.module.scss';

const Layout = ({ children }) => (
  <div className={styles.layout}>{children}</div>
);

Layout.propTypes = {
  children: PropTypes.any
};

export default React.memo(Layout);
