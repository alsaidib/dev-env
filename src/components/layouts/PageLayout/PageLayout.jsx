import React from 'react';

import { layout } from './style.module.scss';

const Layout = ({ children }) => <div className={layout}>{children}</div>;

export default React.memo(Layout);
