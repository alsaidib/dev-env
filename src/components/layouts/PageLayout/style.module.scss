.layout {
	background-color: #eee;
	padding: 10px;
	box-sizing: border-box;
	width: 100%;
	height: 100%;
	position: absolute;
	display: flex;
}
