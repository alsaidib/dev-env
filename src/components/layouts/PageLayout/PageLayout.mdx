---
name: PageLayout
menu: Components
route: /components/PageLayout
---
import PageLayout from './PageLayout'
# Page Layout
The layout whole page which should be wrapped for all components (in Router)
<PageLayout>
  Your App with routes should be here
</PageLayout>

<PropsTable of={PageLayout} />
