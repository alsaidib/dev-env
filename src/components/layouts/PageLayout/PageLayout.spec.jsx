import React from 'react';
import { create } from 'react-test-renderer';
import PageLayout from './PageLayout';

describe('Button component', () => {
  it('render correctly Spinner component', () => {
    const PageLayoutComponent = create(<PageLayout>MY_CHILD_TEXT</PageLayout>);
    expect(PageLayoutComponent).toMatchSnapshot();
  });

  test('it shows the expected text when rendered', () => {
    const PageLayoutComponent = create(<PageLayout>MY_CHILD_TEXT</PageLayout>);
    const PageLayoutComponentInstance = PageLayoutComponent.root;
    expect(PageLayoutComponentInstance.props.children).toBe('MY_CHILD_TEXT');
  });
});
