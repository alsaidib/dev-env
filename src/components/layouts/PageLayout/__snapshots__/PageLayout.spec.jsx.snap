// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render correctly component 1`] = `
<div
  className="layout"
>
  MY_CHILD_TEXT
</div>
`;
